### DISCUSSION TITLE
Unused shellcheck and shfmt

### DISCUSSION THREAD

**Discussion by loiccoyle** [LazyVim/LazyVim#315] [Q&A] [answered]
The `mason` spec at:

 https://github.com/LazyVim/LazyVim/blob/a3aeb27cbc2fa1f19d525b97ebb4cc1a56c7c538/lua/lazyvim/plugins/lsp/init.lua#L157 

installs `shfmt` and `shellcheck` but as far as I can tell they are never setup in `null-ls`.

I guess the `null-ls` spec is missing some sources? Something like:

```lua
nls.builtins.formatting.shfmt
nls.builtins.diagnostics.shellcheck,
nls.builtins.code_actions.shellcheck,
```

**Comment by loiccoyle** [comment]
Actually instead of the `diagnostics.shellcheck` maybe it's best to install the `bash-language-server` (`bashls`) through the `lspconfig` spec instead?

**Reply by folke** [reply]
Yep, you're right! Need to think about this. Probably best to add bash lsp by default instead

**Reply by loiccoyle** [reply]
@folke Have you made up your mind? Sorry for the ping, just making sure you didn't forget :smile:

**Comment by folke** [comment, marked as answer]
I removed shellfmt and shellcheck. Better to install bashls indeed (that's what I had in my config)

**Reply by loiccoyle** [reply]
I saw you also removed `shfmt` maybe its worth ading it to the `null-ls` spec as I don't think  `bashls` has formatting, at least I can't seem to get it to format?

**Reply by folke** [reply]
Added shfnt to null-ls and mason
